lib/facility.rb

```diff
@@ -1,14 +1,91 @@
+require 'date'
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name,
+              :address,
+              :phone, 
+              :services,
+              :registered_vehicles,
+              :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(facility_details)
+    @name = facility_details[:name]
+    @address = facility_details[:address]
+    @phone = facility_details[:phone]
     @services = []
   end
-
-  def add_services(service)
+  
+  def add_service(service)
+    self.collected_fees_helper
```

Comment:
(this applies to many lines of your code)
In Ruby, the use of `self` within a method is implied. This could be
```
def add_service(service)
  collected_fees_helper
  ...
end
```